ApiClient/Source/Query.swift

```diff
@@ -27,12 +27,18 @@ import Foundation
 import Alamofire
 
 public protocol Query {
+    /// The type returned by the query.
     associatedtype Result
     
+    /// The url of the query.
     var url: URL { get }
+    /// The HTTP method of the query.
     var method: HTTPMethod { get }
+    /// The HTTP parameters of the query.
     var parameters: Parameters { get }
+    /// The HTTP headers for the query.
```

Comment:
Please also give an example of a common case, i.e. a query which expects a JSON response.